packages/accounts/src/light-account/utils.ts

```diff
@@ -15,44 +16,100 @@ import {
   polygonAmoy,
   polygonMumbai,
   sepolia,
+  type AccountVersionAddresses,
+  type EntryPointVersion,
+  type OneOf,
   type SmartContractAccount,
 } from "@alchemy/aa-core";
 import { fromHex, type Address, type Chain } from "viem";
 
-export type LightAccountVersion =
-  /** @deprecated This version does not support 1271 signature validation */
-  | "v1.0.1"
-  /** @deprecated This version has a known issue with 1271 validation, it is recommended to use v1.1.0 */
-  | "v1.0.2"
-  | "v1.1.0";
-
-export const LightAccountVersions: Record<
-  LightAccountVersion,
-  {
-    factoryAddress: Address;
-    implAddress: Address;
-  }
+export type LightAccountType = keyof ILightAccountVersions;
+
+export type LightAccountVersionDef<
+  TEntryPointVersion extends EntryPointVersion
 > = {
+  factoryAddress: Address;
+  implAddress: Address;
+  entryPointVersion: OneOf<TEntryPointVersion, EntryPointVersion>;
+};
+
+export interface ILightAccountVersions {
+  LightAccount: {
+    "v1.0.1": LightAccountConfig<"v1.0.1", "0.6.0">;
+    "v1.0.2": LightAccountConfig<"v1.0.2", "0.6.0">;
```

Comment:
we should still mark these as `deprecated` because they shouldn't be used going forward and we should signal to devs to migrate off of them (via upgrades)